internal/commands/tunnel.go

```diff
@@ -221,277 +157,145 @@ func (c *tunnelCmd) SSHKeyEnsurePresent() error {
-
-	ssmSvc := ssm.New(sess)
+	hosts := re.FindAllStringSubmatch(
+		strings.Join(config.SSHForwardConfig.Value, "\n"),
+		-1,
+	)
 
-	out, err := ssmSvc.GetParameter(&ssm.GetParameterInput{
-		Name:           aws.String(fmt.Sprintf("/%s/terraform-output", "dev")),
-		WithDecryption: aws.Bool(true),
-	})
-	if err != nil {
-		pterm.Error.Printfln("Getting tunnel status command")
+	if len(hosts) == 0 {
+		pterm.Error.Printfln("Getting SSH forward config")
 		return err
 	}
 
-	var value []byte
+	pterm.Success.Printfln("Getting SSH forward config")
 
-	value, err = base64.StdEncoding.DecodeString(*out.Parameter.Value)
-	if err != nil {
-		pterm.Error.Printfln("Getting tunnel status command")
-		return err
+	input := &ssm.StartSessionInput{
+		DocumentName: aws.String("AWS-StartPortForwardingSession"),
+		Parameters: map[string][]*string{
+			"portNumber":      {aws.String(strconv.Itoa(22))},
+			"localPortNumber": {aws.String("30022")},
+		},
+		Target: aws.String("i-03a1e68e16db39dea"),
 	}
 
-	var tOut terraformOutput
-
-	err = json.Unmarshal(value, &tOut)
+	out, err := svc.StartSession(input)
 	if err != nil {
-		pterm.Error.Printfln("Getting tunnel status command")
+		pterm.Error.Printfln("Start session")
 		return err
 	}
 
-	command := strings.Split(tOut.Cmd.Value.Tunnel.Status, " ")
-	command = append(command, "-F")
-	command = append(command, fmt.Sprintf("%s/ssh.config", viper.GetString("ENV_DIR")))
-
-	pterm.Success.Printfln("Getting tunnel status command")
+	pterm.Success.Printfln("Start session")
 
-	err = CallProcess(command[0], command[1:])
+	err = ssmsession.NewSSMPluginCommand(c.config.AwsRegion).Forward(out, input)
 	if err != nil {
-		exiterr := err.(*exec.ExitError)
-		status := exiterr.Sys().(syscall.WaitStatus)
-		if status.ExitStatus() != 255 {
-			pterm.Error.Printfln("Running tunnel down command")
-			return err
-		}
+		fmt.Println(err)
+		pterm.Error.Printfln("Forward server to localhost")
 	}
 
-	pterm.Success.Printfln("Running tunnel status command")
-
-	return nil
-}
-
-func (c *tunnelCmd) BastionSHHTunnelDown() error {
-	sess, err := utils.GetSession(&utils.SessionConfig{
-		Region:  c.config.AwsRegion,
-		Profile: c.config.AwsProfile,
-	})
-	if err != nil {
-		pterm.Error.Printfln("Getting AWS session")
-		return err
-	}
+	pterm.Success.Printfln("Forward server to localhost")
 
-	pterm.Success.Printfln("Getting AWS session")
+	for _, h := range hosts {
+		destinationHost := h[2] + ":" + h[3]
 
-	ssmSvc := ssm.New(sess)
+		localPort := h[1]
 
-	out, err := ssmSvc.GetParameter(&ssm.GetParameterInput{
-		Name:           aws.String(fmt.Sprintf("/%s/terraform-output", "dev")),
-		WithDecryption: aws.Bool(true),
-	})
-	if err != nil {
-		pterm.Error.Printfln("Getting tunnel down command")
-		return err
-	}
+		tunnel := sshtunnel.NewSSHTunnel(
+			"ubuntu@localhost",
+			sshtunnel.PrivateKeyFile(privateKeyPath()),
+			destinationHost,
+			localPort,
+		)
 
-	var value []byte
+		tunnel.Server.Port = 30022
```

Comment:
I would use a random port, so it would work in all cases, even if 30022 is used locally. Maybe this would help at least to detect a free port https://www.example-code.com/golang/socket_listen_random_unused_port.asp